bcos-executor/src/precompiled/TablePrecompiled.cpp

```diff
@@ -160,6 +160,34 @@ static void processEntryByValueCond(
     const std::shared_ptr<executor::TransactionExecutive>& _executive, const std::string& tableName,
     std::optional<storage::Condition> keyCondition,
     std::optional<precompiled::Condition> valueCondition, Functor&& processEntry)
+{
+    size_t totalCount = 0;
+    size_t singleCountByKeyMax = keyCondition->getLimit().second;
+    size_t singleCountByKey = singleCountByKeyMax;
+    auto [valueLimitOffset, valueLimitCount] = valueCondition->getLimit();
+    // default condition [ key > "" ]
+    keyCondition->GT("");
+    while (singleCountByKey >= singleCountByKeyMax && totalCount < valueLimitCount)
+    {
+        auto tableKeyList = _executive->storage().getPrimaryKeys(tableName, keyCondition);
+        auto [validCount, entrySize] = processEntry(tableKeyList, valueCondition);
+        singleCountByKey = tableKeyList.size();
+        totalCount += entrySize;
+        valueLimitOffset = std::max(int(valueLimitOffset - validCount), 0);
+        valueCondition->limit(valueLimitOffset, valueLimitCount - totalCount);
+        if (!tableKeyList.empty())
+        {
+            // update condition [ key > tableKeyList.back() ]
+            keyCondition->conditions().back().value = tableKeyList.back();
```

Comment:
the limit of the key condition is still working, why don't clear the limit?
If not clear limit , the start offset will make the result wrong?